src/Sentry/Platforms/Cocoa/CocoaProfiler.cs

```diff
@@ -0,0 +1,62 @@
+using Sentry.Extensibility;
+using Sentry.Internal;
+using Sentry.Cocoa.Extensions;
+using Sentry.Cocoa.Facades;
+
+namespace Sentry.Cocoa;
+
+internal class CocoaProfiler : ITransactionProfiler
+{
+    private readonly SentryOptions _options;
+    private readonly SentryId _traceId;
+    private readonly CocoaSdk.SentryId _cocoaTraceId;
+    private readonly ulong _startTimeNs;
+    private ulong _endTimeNs;
+    private readonly SentryStopwatch _stopwatch = SentryStopwatch.StartNew();
+
+    public CocoaProfiler(SentryOptions options, ulong startTimeNs, SentryId traceId, CocoaSdk.SentryId cocoaTraceId)
+    {
+        _options = options;
+        _startTimeNs = startTimeNs;
+        _traceId = traceId;
+        _cocoaTraceId = cocoaTraceId;
+        _options.LogDebug("Trace {0} profile start timestamp: {1} ns", _traceId, _startTimeNs);
+    }
+
+    /// <inheritdoc />
+    public void Finish()
+    {
+        if (_endTimeNs == 0)
+        {
+            _endTimeNs = _startTimeNs + (ulong)_stopwatch.ElapsedNanoseconds;
+            _options.LogDebug("Trace {0} profile end timestamp: {1} ns", _traceId, _endTimeNs);
+        }
+    }
+
+    public ISerializable? Collect(Transaction transaction)
+    {
+        // TODO change return type of CocoaSDKs CollectProfileBetween to NSMutableDictionary
+        var payload = SentryCocoaHybridSdk.CollectProfileBetween(_startTimeNs, _endTimeNs, _cocoaTraceId)?.MutableCopy() as NSMutableDictionary;
+        if (payload is null)
+        {
+            _options.LogWarning("Trace {0} collected profile payload is null", _traceId);
+            return null;
+        }
+        _options.LogDebug("Trace {0} profile payload collected", _traceId);
+
+        payload["timestamp"] = transaction.StartTimestamp.ToString("o", CultureInfo.InvariantCulture).ToNSString();
+
+        var payloadTx = payload["transaction"]?.MutableCopy() as NSMutableDictionary;
+        if (payloadTx is null)
+        {
+            _options.LogWarning("Trace {0} collected profile payload doesn't have transaction information", _traceId);
+            return null;
+        }
+
```

Comment:
I can't spot if we need the timestamp earlier. Since we bail in case payload is null, move it down?
```suggestion
        var payloadTx = payload["transaction"]?.MutableCopy() as NSMutableDictionary;
        if (payloadTx is null)
        {
            _options.LogWarning("Trace {0} collected profile payload doesn't have transaction information", _traceId);
            return null;
        }

        payload["timestamp"] = transaction.StartTimestamp.ToString("o", CultureInfo.InvariantCulture).ToNSString();
```

Reply:
Sure. I mean it's not going to be null unless sentry-cocoa changes, but I don't have a problem with the change.